### PR TITLE
363 Search endpoint

### LINKED ISSUE #363: Add "Search by subscriber" option
We should have an option to easily find all subscriptions that given endpoint is subscribed to. For example i have a `my-service.com` endpoint and i need to know what kind of events it listens to (i'm not an owner). I would like to have a search option for this, both via API (can be via HermesQL) and Hermes Console.

### PR DESCRIPTION
**Pull request by ingwarsw**
- Query endpoint for all data
- New LIKE operator
- Closes #363

### CODE REVIEW

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/group/GroupService.java [hunk 1]
```diff
@@ -3,16 +3,18 @@
...
+import pl.allegro.tech.hermes.common.query.Query;
 import pl.allegro.tech.hermes.domain.group.GroupNotExistsException;
 import pl.allegro.tech.hermes.domain.group.GroupRepository;
 import pl.allegro.tech.hermes.infrastructure.MalformedDataException;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
-@Service
+@Component
```
Comment (chemicL) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/group/GroupService.java]: any reason for changing to @Component?

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/query/matcher/LikeMatcher.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            this.expected = Pattern.compile(asString(expected));
+        } catch (PatternSyntaxException e) {
+            throw new MatcherException(String.format("Could not parse regexp pattern: '%s'", expected), e);
+        }
+    }
+
+    @Override
+    public boolean match(Object value) {
+        try {
+            if (expected == null) {
```
Comment (chemicL) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/query/matcher/LikeMatcher.java]: This doesn't seem to be possible

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/group/GroupService.java (hunk 1 above)
Comment (ingwarsw) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/group/GroupService.java]: Yep.. to match all other Services..

Maybe all should be change do @Service.. but they should use the same annotation.

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/management/QueryEndpointTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        management.group().create(new Group("testGroup1", "Owner1", "Support3", "Contact1"));
+        management.group().create(new Group("testNot1", "Owner2", "Support3", "Contact1"));
+        management.group().create(new Group("testGroup2", "Owner2", "Support4", "Contact2"));
+        management.group().create(new Group("testGroup3", "Owner1", "Support2", "Contact1"));
+
+        // when
+        List<Group> found = management.query().queryGroup(query);
+
+        // then
+        assertThat(found).hasSize(resultSize);
```
Comment (chemicL) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/management/QueryEndpointTest.java]: This feels neat, but I'm reluctant to admit it feels correct. Can we also test whether the result matches specific 'ids' ?

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/group/GroupService.java (hunk 1 above)
Comment (chemicL) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/group/GroupService.java]: Absolutely, thanks for pointing that out. IMHO @Service should be the way to go, but that's just polishing ;-)

**Comment by adamdubiel** [comment]
squash and we are all set :+1:

**Comment by chemicL** [comment]
:+1:

**Pull request merged**